ggdeploymentd/src/deployment_handler.c

```diff
@@ -1977,9 +1990,188 @@ static GglError wait_for_deployment_status(GglMap resolved_components) {
+            GGL_LOGE("Reset update did not parse into a list "
+                     "during configuration updates.");
+            return GGL_ERR_INVALID;
+        }
+        GGL_LIST_FOREACH(reset_element, reset_configuration->list) {
+            if (reset_element->type != GGL_TYPE_BUF) {
+                GGL_LOGE("Configuration key for reset config "
+                         "update not provided as a buffer.");
+                return GGL_ERR_INVALID;
+            }
+            ret = ggl_gg_config_delete(GGL_BUF_LIST(
+                GGL_STR("services"),
+                component_name,
+                GGL_STR("configuration"),
+                reset_element->buf
+            ));
+
+            if (ret != GGL_ERR_OK) {
+                GGL_LOGE(
+                    "Failed to perform configuration reset updates "
+                    "for component %.*s.",
+                    (int) component_name.len,
+                    component_name.data
+                );
+                return ret;
+            }
+
+            GGL_LOGI(
+                "Made a configuration reset update for component "
+                "%.*s",
+                (int) component_name.len,
+                component_name.data
+            );
+        }
+    }
+
+    return GGL_ERR_OK;
+}
+
+static GglError apply_merge_config(
+    GglBuffer component_name, GglMap component_config_map
+) {
+    GglObject *merge_configuration = NULL;
+    GglError ret = ggl_map_validate(
+        component_config_map,
+        GGL_MAP_SCHEMA(
+            { GGL_STR("merge"), false, GGL_TYPE_MAP, &merge_configuration }
+        )
+    );
+    if (ret != GGL_ERR_OK) {
+        return ret;
+    }
+
+    // If there is no merge configuration, then there is no
+    // configuration update to make
+    if (merge_configuration != NULL) {
+        if (merge_configuration->type != GGL_TYPE_MAP) {
+            GGL_LOGE("Merge update did not parse into a map during "
+                     "configuration updates.");
+            return GGL_ERR_INVALID;
+        }
+
+        // TODO: Use deployment timestamp not the current timestamp
+        // after we support deployment timestamp
+        ret = ggl_gg_config_write(
+            GGL_BUF_LIST(
+                GGL_STR("services"), component_name, GGL_STR("configuration")
+            ),
+            *merge_configuration,
+            NULL
+        );
+
+        if (ret != GGL_ERR_OK) {
+            GGL_LOGE(
+                "Failed to write configuration merge updates for "
+                "component %.*s to ggconfigd.",
+                (int) component_name.len,
+                component_name.data
+            );
+            return ret;
+        }
+
+        GGL_LOGI(
+            "Made a configuration merge update for component %.*s",
+            (int) component_name.len,
+            component_name.data
+        );
+    }
+
+    return GGL_ERR_OK;
+}
+
+static GglError apply_configurations(
```

Comment:
Can we move this method and the merge and reset methods into a separate file, maybe something like `config_update_handler`? Adding to deployment handler makes it more difficult to manage, this will save us from some refactoring in the future.